src/components/NcAppSidebarTab/NcAppSidebarTab.vue

```diff
@@ -130,7 +130,7 @@ export default {
 		 * @return {import('vue').VNode[]}
 		 */
 		renderIcon() {
-			return this.$scopedSlots.icon?.()
+			return this.$slots.icon
```

Comment:
Otherwise dynamic component slot will have no effect. But I haven't tested it.

```suggestion
			return defineComponent((props, { slots }) => this.$slots.icon?.() || slots.default())
```

Reply:
Also, `this.$slots.icon` is a render function, so we need to rename a method and update JSDoc.

Reply:
The icon slot is reactive, if that is what you mean:

[Bildschirmaufzeichnung vom 2023-11-17, 09-20-19.webm](https://github.com/nextcloud-libraries/nextcloud-vue/assets/2496460/df97b015-d537-4d51-9147-cc7b8cd28344)


```vue
<template>
		<NcCheckboxRadioSwitch v-model:checked="iconType" value="magnify" name="icon_radio" type="radio">Magnify</NcCheckboxRadioSwitch>
		<NcCheckboxRadioSwitch v-model:checked="iconType" value="cog" name="icon_radio" type="radio">Cog</NcCheckboxRadioSwitch>
	<NcAppSidebar
		:starred="starred"
		name="cat-picture.jpg"
		subname="last edited 3 weeks ago">
		<NcAppSidebarTab name="Search" id="search-tab">
			<template #icon>
				<Cog v-if="iconType === 'cog'"  :size="20" />
				<Magnify v-else :size="20" />
			</template>
			Search tab content
		</NcAppSidebarTab>
		<NcAppSidebarTab name="Settings" id="settings-tab">
			<template #icon>
				<Cog :size="20" />
			</template>
			Settings tab content
		</NcAppSidebarTab>
		<NcAppSidebarTab name="Sharing" id="share-tab">
			<template #icon>
				<ShareVariant :size="20" />
			</template>
			Sharing tab content
		</NcAppSidebarTab>
	</NcAppSidebar>
</template>
<script>
	import Magnify from 'vue-material-design-icons/Magnify'
	import Cog from 'vue-material-design-icons/Cog'
	import ShareVariant from 'vue-material-design-icons/ShareVariant'

	export default {
		components: {
			Magnify,
			Cog,
			ShareVariant,
		},
		data() {
			return {
				starred: false,
				iconType: 'magnify',
			}
		},
		methods: {
			figureClick() {
				alert('figure clicked')
			},
		},
	}
</script>
```

Reply:
Is it ok to leave it as `return this.$slots.icon` or did I misunderstood your concern? If yes, then I would just change the function name to `renderFunctionIcon()` and adjust the JSDoc. Could you give me a hint, what the correct type would be?

Reply:
> The icon slot is reactive, if that is what you mean:

Before, `VNodes` component was used, which not only rendered vnodes from from `vnodes` prop but also rendered its default content as a fallback content.

It was used to render `class-based` icon, when `icon` prop is provided instead of the `#icon` slot.

Now it is replaced with dynamic component `<component :is="render function from $slots.icon">`. Which renders vnodes from `renderIcon = $slots.icon`, but doesn't render its default slot.

So, `icon` prop doesn't work.

Reply:
> `class-based` icon

I think we can finally get rid of icon classes with nextcloud-vue 9, no?

Reply:
> > `class-based` icon
> 
> 
> 
> I think we can finally get rid of icon classes with nextcloud-vue 9, no?

That would be nice. But let's not break it here. We can simply keep using `VNodes` for now.
Then we have a working version now and can still break it later.

Reply:
Fixed it by bringing back `VNodes`.